sdks/java/core/src/main/java/org/apache/beam/sdk/io/CompressedSource.java

```
@@ -149,6 +150,26 @@ public ReadableByteChannel createDecompressingChannel(ReadableByteChannel channe
         return Channels.newChannel(
             new BZip2CompressorInputStream(Channels.newInputStream(channel)));
       }
+    },
+
+    /**
+     * Reads a byte channel assuming it is compressed with zip.
+     */
+    ZIP {
+      @Override
+      public boolean matches(String fileName) {
+        return fileName.toLowerCase().endsWith(".zip");
+      }
+
+      public ReadableByteChannel createDecompressingChannel(ReadableByteChannel channel)
+        throws IOException {
+        ZipInputStream zip = new ZipInputStream(Channels.newInputStream(channel));
+        // TODO deal with multi-entries zip
+        if (zip.getNextEntry() != null) {
+          return Channels.newChannel(zip);
+        }
+        return null;
```

should you instead throw an `UnsupportedOperationException` explaining the bug?

And maybe file a related JIRA?